### ISSUE TITLE
[CI] S3BlobStoreRepositoryTests testMetrics failing

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#101608]
**Build scan:**
https://gradle-enterprise.elastic.co/s/cmzydsjar4s3c/tests/:modules:repository-s3:internalClusterTest/org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests/testMetrics
**Reproduction line:**
```
./gradlew ':modules:repository-s3:internalClusterTest' --tests "org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testMetrics" -Dtests.seed=56B8E75E80922D01 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=bg-BG -Dtests.timezone=America/Kentucky/Louisville -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests&tests.test=testMetrics

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <10L>
     but: was <9L>

  at __randomizedtesting.SeedInfo.seed([56B8E75E80922D01:A8AA35FC9C985AB2]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.lambda$testMetrics$3(S3BlobStoreRepositoryTests.java:285)
  at java.util.ArrayList.forEach(ArrayList.java:1596)
  at org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testMetrics(S3BlobStoreRepositoryTests.java:278)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

**Comment by cbuescher** [comment]
And [another one from today](https://gradle-enterprise.elastic.co/s/hniaialqw5k34/tests/task/:modules:repository-s3:internalClusterTest/details/org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests/testMetrics?top-execution=1)

**Comment by ywangd** [comment]
Relabel this to lwo-risk since it is an off-by-one error in metric number comparison which is not a critical path.

**Comment by fcofdez** [comment]
I think this is likely a duplicate of https://github.com/elastic/elasticsearch/issues/88841

**Comment by DaveCTurner** [comment]
Improved the test output on failure in #102386 and #102387, now waiting on another failure to confirm.

**Comment by DaveCTurner** [comment]
https://gradle-enterprise.elastic.co/s/cxskzugh74ktq/tests/task/:modules:repository-s3:internalClusterTest/details/org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests/testMetrics looks like another failure?

**Comment by ywangd** [comment]
Unfortunately the AWS debug logging was [disabled](https://gradle-enterprise.elastic.co/s/cxskzugh74ktq/tests/task/:modules:repository-s3:internalClusterTest/details/org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests/testMetrics?expanded-stacktrace=WyIwIl0&focused-execution=1&top-execution=1#L0) due to #105020. I raised #109068 to reenable it. I'll ask core-infra whether it is possible to skip logger checking for tests.

**Comment by ywangd** [comment]
It still has not failed yet since May 28.

**Comment by DaveCTurner** [comment]
> It still has not failed yet since May 28.

Perhaps because it has been muted since then, see 520a1599a65301c0cac44afe1ea306d3f718416f 🤦 I opened https://github.com/elastic/elasticsearch/pull/114129 to start running the test again.

**Comment by pxsalehi** [comment]
I've been running this test over the past couple of days with stress-ng on and off randomly. over 20k+ runs and no failure. IMO, we can close it since it doesn't reproduce.

**Comment by ywangd** [comment]
Since last update (2 month ago), there [has _not_ been any actual CI failure](https://es-delivery-stats.elastic.dev/app/r/s/9fKbO). The only failure is unrelated and due to [testing](https://github.com/elastic/elasticsearch/pull/117863) of upload checksum.

**Comment by ywangd** [comment]
No new failure. Keep waiting ...

**Comment by nicktindall** [comment]
This hasn't failed in the last month, I think that means it's been 4 months since it failed. I'm inclined to close this?

Failures in last month: https://es-delivery-stats.elastic.dev/app/r/s/uOoHS

**Comment by DaveCTurner** [comment]
With the AWS SDK now upgraded to v2 in #126843 I believe this test failure is now either gone or, at the very least, changed beyond all recognition, so I'm closing it.
